src/devices/cpu/dsp56156/inst.h

```diff
@@ -1452,15 +1389,10 @@ class Jcc: public Instruction
 		decode_cccc_table(BITSn(word0,0x000f), m_mnem);
 		return true;
 	}
-	void disassemble(std::string& retString) const override
+	std::string disassemble() const override
 	{
-		std::string opcode = "j" + opMnemonicAsString(m_mnem);
-		// NEW // sprintf(opcode_str, "j.%s", M);
-
-		char temp[32];
-		sprintf(temp, ">$%x", m_displacement);
-		// NEW // sprintf(temp, "$%04x", word1);
-		retString = opcode + " " + std::string(temp);
+		auto opcode = "j" + opMnemonicAsString(m_mnem);
+		return opcode + util::string_format(" >$%x", m_displacement);
```

Comment:
Why use `auto` here?  The type isn’t repeated tautologically in the expression, and it doesn’t make the code clearer to use `auto`.  But if you’re changing all the lines anyway, why not turn it into a single `util::string_format` and avoid the extra concatenation and temporary?

Reply:
I'm an almost always auto kind of guy.
That aside, having the 'opcode' temporary is arguably more readable code, whilst ignoring performance.
But happy to update

Reply:
I'm not sure if you're against string concatenation due to concerns over performance.
I've done a bit of benchmarking for reference. This was a made up (but representable) function driven by random numbers called 100,000 times:

```
  // String concat
  return opcode + " " + reg1 + " " + util::string_format("%4.2f %4.2f %4.2f", x, y, z) + "\n";
```
VS
```
  // Fully string_format
  return util::string_format("%s %s %4.2f %4.2f %4.2f\n", opcode, reg1, x, y, z);
```

On a Mac, the concat method is about 10% quicker than putting everything inside the string_format.

However both are very fast (compared with say printing the result), so I think the choice of style should be based on source readability. I'm happy to accept in some cases the fully string_format could be more readable so happy with either option. I just wanted to make sure we choose it for the right reasons.